Sources/RSocketCore/Channel Handler/ConnectionStreamHandler.swift

```diff
@@ -14,10 +14,53 @@
  * limitations under the License.
  */
 
+import Foundation
 import NIO
 
-final class ConnectionStreamHandler: ChannelInboundHandler {
-    typealias InboundIn = FrameBody
+final class ConnectionStreamHandler {
+    private var keepAliveHandle: RepeatedTask?
+    private var isKeepAliveStarted = false
+    private var lastReceivedTime = Date()
+    private let setup: ClientSetupConfig
+
+    internal init(config: ClientSetupConfig) {
+        setup = config
+    }
+}
+
+extension ConnectionStreamHandler: ChannelInboundHandler {
+    typealias InboundIn = Frame
     typealias OutboundOut = FrameBody
-    // TODO
+
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        let frame = unwrapInboundIn(data)
+        if frame.header.type == .keepalive {
+            lastReceivedTime = Date()
+            if frame.header.flags.contains(.keepAliveRespond) {
+                let responseKeepAliveFrameBody = KeepAliveFrameBody(respondWithKeepalive: false, lastReceivedPosition: 0, data: Data())
+                let keepAliveFrame = Frame(header: responseKeepAliveFrameBody.header(), body: responseKeepAliveFrameBody.body())
+                context.channel.writeAndFlush(keepAliveFrame, promise: nil)
+            }
+        }
+    }
+
+    func channelActive(context: ChannelHandlerContext) {
+        guard !isKeepAliveStarted, setup.timeBetweenKeepaliveFrames > 0 else { return }
+        lastReceivedTime = Date()
+        keepAliveHandle = context.eventLoop.scheduleRepeatedAsyncTask(initialDelay: .milliseconds(Int64(setup.timeBetweenKeepaliveFrames)), delay: .milliseconds(Int64(setup.timeBetweenKeepaliveFrames))) { task in
+            if Int32(self.lastReceivedTime.timeIntervalSince(Date()) * 1000) > self.setup.maxLifetime {
+                let errorFrame = Frame(header: FrameHeader(streamId: .connection, type: FrameType.error, flags: FrameFlags()), body: FrameBody.error(ErrorFrameBody(error: Error.connectionClose(message: "KeepAlive timeout exceeded"))))
```

Comment:
You can create a `Frame` from an `Error` by just calling `asFrame(withStreamId:)`
```suggestion
                let errorFrame = Error.connectionClose(message: "KeepAlive timeout exceeded").asFrame(withStreamId: .connection)
```